src/main/java/org/folio/roles/service/permission/PermissionOverrider.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.roles.service.permission;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.ws.rs.NotFoundException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Map.Entry;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.common.utils.permission.model.PermissionAction;
+import org.folio.common.utils.permission.model.PermissionData;
+import org.folio.common.utils.permission.model.PermissionType;
+import org.folio.roles.configuration.property.PermissionMappingProperties;
+import org.folio.roles.exception.ServiceException;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Service;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class PermissionOverrider {
+
+  private final ObjectMapper objectMapper;
+  private final ResourceLoader resourceLoader;
+  private final PermissionMappingProperties permissionMappingProperties;
+
+  public Map<String, PermissionData> getPermissionMappings() {
+    var sourcePath = permissionMappingProperties.getSourcePath();
+    try {
+      var resource = resourceLoader.getResource(sourcePath);
+      if (resource.exists()) {
+        var permissions =
+          objectMapper.readValue(resource.getInputStream(), new TypeReference<Map<String, Permission>>() {});
+        return permissions.entrySet().stream()
+          .collect(toMap(Map.Entry::getKey, PermissionOverrider::mapToPermissionData));
+      } else {
+        log.warn("Source not found: {}", sourcePath);
+        throw new NotFoundException("Source not found: " + sourcePath);
```

Comment:
I wouldn't use REST-related exception, `jakarta.ws.rs.NotFoundException`, in this case. Resource usually throws IOException or some of its descendants, like `FileNotFoundException`

Reply:
exception type changed